src/Roslyn.Diagnostics.Analyzers/Core/DoNotCopyValue.cs

```diff
@@ -0,0 +1,1443 @@
+            isEnabledByDefault: true,
+            description: s_localizableNoUnboxingDescription,
+            helpLinkUri: null,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule, UnsupportedUseRule, NoBoxingRule, NoUnboxingRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var cache = new NonCopyableTypesCache(context.Compilation);
+                context.RegisterOperationBlockAction(context => AnalyzeOperationBlock(context, cache));
+            });
+        }
+
+        private static void AnalyzeOperationBlock(OperationBlockAnalysisContext context, NonCopyableTypesCache cache)
+        {
+            var walker = new NonCopyableWalker(context, cache);
+            foreach (var operation in context.OperationBlocks)
+            {
+                walker.Visit(operation);
+            }
+        }
+
+        private static VisitReleaser<T> TryAddForVisit<T>(HashSet<T> set, T? value, out bool added)
+            where T : class
+        {
+            if (value is null)
+            {
+                added = false;
+                return default;
+            }
+
+            added = set.Add(value);
+            if (!added)
+                return default;
+
+            return new VisitReleaser<T>(set, value);
+        }
+
+        [SuppressMessage("Performance", "CA1815:Override equals and operator equals on value types", Justification = "This type is never compared.")]
+        private readonly struct VisitReleaser<T> : IDisposable
+            where T : class
+        {
+            private readonly HashSet<T> _set;
+            private readonly T _value;
+
+            public VisitReleaser(HashSet<T> set, T value)
+            {
+                _set = set;
+                _value = value;
+            }
+
+            public void Dispose()
+            {
+                _set?.Remove(_value);
+            }
+        }
+
+        private sealed class NonCopyableWalker : OperationWalker
+        {
+            private readonly OperationBlockAnalysisContext _context;
+            private readonly NonCopyableTypesCache _cache;
+            private readonly HashSet<IOperation> _handledOperations = new HashSet<IOperation>();
+
+            public NonCopyableWalker(OperationBlockAnalysisContext context, NonCopyableTypesCache cache)
+            {
+                _context = context;
+                _cache = cache;
+            }
+
+            public override void VisitAddressOf(IAddressOfOperation operation)
+            {
+                CheckTypeInUnsupportedContext(operation);
```

Comment:
Do you override every single method on `OperationWalker`, with this being the default implementation?

Reply:
Yes. I do that until I manually update the method with a more complete implementation.